src/ui/widget-list-page/WidgetListPage.js

```diff
@@ -0,0 +1,59 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import InternalPage from 'ui/internal-page/InternalPage';
+import WidgetListTable from 'ui/widget-list-page/WidgetListTable';
+import RowListContainer from 'ui/widget-list-page/RowListContainer';
+import PageTitle from 'ui/internal-page/PageTitle';
+
+import { Row, Col, Button } from 'patternfly-react';
+
+import { FormattedMessage } from 'react-intl';
+import 'sass/widget-list-page/WidgetListPage.scss';
+import { formattedText } from 'frontend-common-components';
+
+const WidgetListPage = ({ onClickCreate, onLoad }) => {
+  const onClickAdd = (ev) => {
+    ev.preventDefault();
+    onClickCreate();
+    // alert('connect to create a new widget function');
+  };
+  onLoad();
```

Comment:
This will be called every time the component renders. We need this to be called only once, when the component is mounted (rendered) for the first time.
Please change this component to a class component, name this prop "onWillMount" and call it in componentWillMount() method